### ISSUE TITLE
No way to preload Image from assets before first builder (needed for smooth splash screen with images)

### ISSUE THREAD

**Issue by filipinascimento** [flutter/flutter#26127]
Hi, 

I'm trying to make a simple splash screen with a background Image. The image is a 1024x1024 png file inside assets. I already changed `LaunchScreen.storyboard` and `LaunchImage` to the same image and same fit. Everything works fine except for the first frame of flutter app (corresponding to the time that the image was not loaded).

I could try to preload the image but I did not find a way to preload the image before having a builder context.
 
## Steps to Reproduce

1. Create new project, change launch image and launchscreen.storyboard in Xcode to show a background image.
2. run the code:

```dart
void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  MyApp({
    Key key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {

    ImageProvider backgroundImage = AssetImage('assets/Background/login.png');
    backgroundImage.resolve(createLocalImageConfiguration(context));

    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData.light(),
      home: Scaffold(
        backgroundColor: Colors.red,
        body: Container(
          decoration: BoxDecoration(
            image: DecorationImage(
              image: backgroundImage,
              fit: BoxFit.fill,
            ),
          ),
          child: Center(child: CircularProgressIndicator()),
        ),
      ),
    );
  }
}
```

In this example the screen flashes red for at least a frame before showing the image correctly. Is there a documented way to preload images from assets before running any builder?

I only tested this on iOS.

Thanks

## Logs

```log
[✓] Flutter (Channel beta, v1.0.0, on Mac OS X 10.14.2 18C54, locale en-US)
    • Flutter version 1.0.0 at /Users/filipi/Dropbox/flutter
    • Framework revision 5391447fae (5 weeks ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/filipi/Library/Android/sdk
    • Android NDK at /Users/filipi/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 31.1.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1136-b06)

[✓] VS Code (version 1.30.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.21.1

[✓] Connected device (2 available)
    • Android SDK built for x86 • emulator-5554                        •
      android-x86 • Android 8.0.0 (API 26) (emulator)
    • iPhone 6s                 • 0C24A047-645B-4E59-85A7-E3EE17EADF52 • ios
      • iOS 12.1 (simulator)

• No issues found!
```

**Comment by momoDragon** [comment]
https://stackoverflow.com/questions/51343735/flutter-image-preload

maybe this?

**Comment by filipinascimento** [comment]
Unfortunately this also does not work as preloadImage also needs a Builder context for createLocalImageConfiguration internally, so the problem persists. If you put this inside the first builder, Launch screen created by system will flip to Flutter before the image is preloaded. There is no way to delay builder with a Future after you are inside it. Or there is an alternative?

**Comment by zoechi** [comment]
https://docs.flutter.io/flutter/widgets/SingleTickerProviderStateMixin/didChangeDependencies.html
should work then. 
(I thought that the `context` property also allows to access the `BuildContext` in `initState` already, but I haven't checked since a while and might mix something up)

**Comment by filipinascimento** [comment]
Hello, thanks for this information.
I tried using didChangeDependencies but had no success in preventing build to run before the image is cached. First I updated myApp class to a `statefuWidget` and override `didChangeDependencies`.

```dart

void main() {
  runApp(MyApp());
}

class MyApp extends StatefulWidget {
  MyApp({
    Key key,
  }) : super(key: key);

  @override
  MyAppState createState() {
    return new MyAppState();
  }
}

class MyAppState extends State<MyApp> {
  ImageProvider backgroundImage;

  @override
  void didChangeDependencies() async{
    backgroundImage = AssetImage('assets/Background/login.png');
    await precacheImage(backgroundImage,context);
    super.didChangeDependencies();
  }

  @override
  Widget build(BuildContext context) {
    
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData.light(),
      home: Scaffold(
        backgroundColor: Colors.red,
        body: Container(
          decoration: BoxDecoration(
            image: DecorationImage(
              image: backgroundImage,
              fit: BoxFit.fill,
            ),
          ),
          child: Center(child: CircularProgressIndicator()),
        ),
      ),
    );
  }
}
```
In this example I'm using an `async` `didChangeDependencies`. However, this function completes after build is called, so no cached image there. If I use a synchronized `didChangeDependencies`, `preloadImage` function, which is async, finishes after the first build. I could not figure out a way to run and wait for anything that is async before the first build.

**Comment by torvalde** [comment]
I ended up with base64 encoding the image and load with Image.memory(), this is ugly but it seems to work for now.

**Comment by maqoo** [comment]
> I ended up with base64 encoding the image and load with Image.memory(), this is ugly but it seems to work for now.

Could you elaborate on this solution a bit? When do you load the image from base64 for instance?

**Comment by DevMA523** [comment]
Could we just display a loading screen when the images are loading to avoid the gap (then display the page) ?

How to check if all images are loaded ?

**Comment by torvalde** [comment]
> > I ended up with base64 encoding the image and load with Image.memory(), this is ugly but it seems to work for now.
> 
> Could you elaborate on this solution a bit? When do you load the image from base64 for instance?

I load and display the base64 encoded image in a fullscreen popup as I get the user info. 

`import 'helpers/startup_image.dart';

....
class .... {
Uint8List image = Base64Decoder().convert(imageString);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        body: Container(
            child: Image.memory(
      image,
      fit: BoxFit.cover,
      height: MediaQuery.of(context).size.height,
      width: MediaQuery.of(context).size.width,
      alignment: Alignment.center,
    )));
  }
}`

This is just a hack to avoid flickering while the background image is loaded.

**Comment by szotp** [comment]
It would be nice to have a special widget, named `Splash` for example, that would display exactly the same pixels as the native launch screen. This would save us from needing to reimplement the splash three times, althought it would only support basic transitions.

**Comment by hardypatel30** [comment]
> > > I ended up with base64 encoding the image and load with Image.memory(), this is ugly but it seems to work for now.
> > 
> > 
> > Could you elaborate on this solution a bit? When do you load the image from base64 for instance?
> 
> I load and display the base64 encoded image in a fullscreen popup as I get the user info.
> 
> ```
> import 'helpers/startup_image.dart';
> 
> ....
> class .... {
> Uint8List image = Base64Decoder().convert(imageString);
> 
>   @override
>   Widget build(BuildContext context) {
>     return Scaffold(
>         body: Container(
>             child: Image.memory(
>       image,
>       fit: BoxFit.cover,
>       height: MediaQuery.of(context).size.height,
>       width: MediaQuery.of(context).size.width,
>       alignment: Alignment.center,
>     )));
>   }
> }```
> 
> This is just a hack to avoid flickering while the background image is loaded.
> ```

Can you show us the way of converting image from asset path itself?

**Comment by hardypatel30** [comment]
Thanks alot for the code.

**Comment by szotp** [comment]
@efreibe main can be async
```dart
Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await loadImage('assets/Background/login.png');
  runApp(MyApp());
}

```

**Comment by efreibe** [comment]
Good idea.

**Comment by szotp** [comment]
@efreibe I think that they fixed it recently, it should display splash until first build method runs.

**Comment by efreibe** [comment]
Ok. I'm using flutter from the stable channel and I even didn't know that
it was resolved.
Good to hear that.

Thank you

**Comment by ldemyanenko** [comment]
In what flutter version this issue was resolved?

**Comment by mleonhard** [comment]
Here's my solution, adapted from @efreibe 's post above.  This provides a smooth transition on Android.  I can report back on iOS once I implement a workaround for #20827.
```dart
import 'dart:async' show Completer;
import 'package:flutter/painting.dart'
    show
        AssetImage,
        ImageConfiguration,
        ImageInfo,
        ImageStream,
        ImageStreamListener;
import 'package:flutter/widgets.dart'
    show
        Alignment,
        BoxConstraints,
        Center,
        Color,
        Container,
        EdgeInsets,
        Image,
        runApp,
        SafeArea,
        StatelessWidget,
        Widget,
        WidgetsFlutterBinding;

class Logo extends StatelessWidget {
  static AssetImage _getAssetImage() => AssetImage('assets/logo.1024x477.png');

  static Future<void> preload() async {
    final ImageStream imageStream =
        _getAssetImage().resolve(ImageConfiguration.empty);
    final Completer completer = Completer();
    final ImageStreamListener listener = ImageStreamListener(
      (ImageInfo imageInfo, bool synchronousCall) => completer.complete(),
      onError: (e, stackTrace) => completer.completeError(e, stackTrace),
    );
    imageStream.addListener(listener);
    return completer.future
        .whenComplete(() => imageStream.removeListener(listener));
  }

  @override
  Widget build(_) => Container(
      // When changing this, also update ios LaunchScreen.storyboard.
      margin: const EdgeInsets.fromLTRB(0, 100, 0, 40),
      constraints: const BoxConstraints.tightFor(width: 200, height: 93),
      child: Image(
        image: _getAssetImage(),
        semanticLabel: 'App Name logo',
      ));
}

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Logo.preload();
  runApp(Container(
      color: Color(0xFFffffff),
      alignment: Alignment.topCenter,
      child: Center(child: Logo())));
}
```

The first version of this initial screen used `SafeArea` and the logo would jump downwards on load on Android.  I found no way to make the Android splash screen inset at the top.  The easiest solution was to stop using `SafeArea` in Flutter and in the iOS splash screen.

**Comment by parisholley** [comment]
@mleonhard your solution of preloading fixed my iOS issue (base64 embed above didn't help).

**Comment by RoughIndustries** [comment]
Just a heads up I solved this by:

`class SplashScreen extends StatefulWidget {
  const SplashScreen(this.appBloc, this.loginBloc);

  @override
  _SplashScreenState createState() => _SplashScreenState();
}

class _SplashScreenState extends State<SplashScreen> {
  Image SplashImage;
  bool cont = false;

//  @override
//  bool get wantKeepAlive => true;

  @override
  void initState() {
    super.initState();
    SplashImage = Image.asset('resources/images/launch/launchimagelarge.png');
  }

  @override
  void dispose() {
    super.dispose();
  }

  @override
  void didChangeDependencies() {
    precacheImage(
      SplashImage.image,
      context,
    ).then((_) {
      setState(() {});
      cont = true;
    });
    super.didChangeDependencies();
  }

  @override
  Widget build(BuildContext context) {
    if (cont) {
        // Did bloc logic here to update the state now that I am loaded
    }
    return Container(
      width: MediaQuery.of(context).size.width,
      height: MediaQuery.of(context).size.height,
      decoration: BoxDecoration(
        image: DecorationImage(
          image: SplashImage.image,
          fit: BoxFit.cover,
        ),
      ),
      child: Container(),
    );
  }
}`

**Comment by searchy2** [comment]
Based on this issues, it seems the consensus is that preloading an image requires a BuildContext.
https://stackoverflow.com/questions/62478972/preload-image-without-buildcontext

**Comment by TahaTesser** [comment]
![ezgif com-resize (1)](https://user-images.githubusercontent.com/48603081/94149739-e42da400-fe95-11ea-83e9-8ec107d57e10.gif)


<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  MyApp({
    Key key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    ImageProvider backgroundImage = AssetImage('assets/bg.jpg');
    backgroundImage.resolve(createLocalImageConfiguration(context));

    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData.light(),
      home: Scaffold(
        backgroundColor: Colors.red,
        body: Container(
          decoration: BoxDecoration(
            image: DecorationImage(
              image: backgroundImage,
              fit: BoxFit.fill,
            ),
          ),
          child: Center(child: CircularProgressIndicator()),
        ),
      ),
    );
  }
}


```

</details> 


<details> 
<summary>flutter doctor -v</summary> 

```bash
✓] Flutter (Channel master, 1.22.0-10.0.pre.340, on Mac OS X 10.15.6 19G2021 x86_64, locale en-GB)
    • Flutter version 1.22.0-10.0.pre.340 at /Users/tahatesser/Code/flutter_master
    • Framework revision 1fef5912cf (6 hours ago), 2020-09-23 21:37:08 -0700
    • Engine revision dd35b5b659
    • Dart version 2.10.0 (build 2.10.0-156.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0, Build version 12A7209
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

[✓] Connected device (5 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios            • iOS 14.0
    • iPhone 11 (mobile)     • 9A8EFBFE-A806-4FF9-ACA5-F3F98D94979D • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • Mac OS X 10.15.6 19G2021 x86_64
    • Web Server (web)       • web-server                           • web-javascript • Flutter Tools
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 85.0.4183.102

• No issues found!

```

</details>

**Comment by fluttercorner** [comment]
[How to set Background Image to Scaffold in Flutter](https://fluttercorner.com/how-to-set-background-image-to-scaffold-in-flutte/)

**Comment by lukepighetti** [comment]
I find it odd that we can precache SVGs for `flutter_svg` and flare assets for `flare_flutter` without build context, but there's currently no way to precache image assets.

**Comment by Maistho** [comment]
I use this method to load images before starting the app, which seems to work as intended:

`main.dart`

```dart
Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await loadImage(AssetImage('assets/background-main.jpg'));
  runApp(MyApp());
}
```

`preload_image.dart`

```dart
// @dart=2.12
import 'dart:async';

import 'package:flutter/foundation.dart';
import 'package:flutter/painting.dart';
import 'package:flutter/services.dart';

Future<void> loadImage(ImageProvider provider) {
  final config = ImageConfiguration(
    bundle: rootBundle,
    devicePixelRatio: 1,
    platform: defaultTargetPlatform,
  );
  final Completer<void> completer = Completer();
  final ImageStream stream = provider.resolve(config);

  late final ImageStreamListener listener;

  listener = ImageStreamListener((ImageInfo image, bool sync) {
    debugPrint("Image ${image.debugLabel} finished loading");
    completer.complete();
    stream.removeListener(listener);
  }, onError: (dynamic exception, StackTrace? stackTrace) {
    completer.complete();
    stream.removeListener(listener);
    FlutterError.reportError(FlutterErrorDetails(
      context: ErrorDescription('image failed to load'),
      library: 'image resource service',
      exception: exception,
      stack: stackTrace,
      silent: true,
    ));
  });

  stream.addListener(listener);
  return completer.future;
}
```

**Comment by dodatw** [comment]
I used @Maistho function to preload image in main, and it really help that can fix flash issue before switch between native and flutter page.
BUT, it have a side-effect that I cannot reproduce at my side.
After publish on store, it have 1% chance crash when load image in main. (from crashlytics)

```
Crashed: io.flutter.1.io
0  libsystem_kernel.dylib         0x1b737884c __pthread_kill + 8
1  libsystem_pthread.dylib        0x1d3c5c9e8 pthread_kill + 212
2  libsystem_c.dylib              0x1941348f4 abort + 100
3  Metal                          0x1a0ec7b84 MTLGetWarningMode + 214
4  IOGPU                          0x1c5fa9df8 -[IOGPUMetalCommandBuffer pushDebugGroup:] + 238
5  Metal                          0x1a0eb84dc -[_MTLCommandEncoder initWithCommandBuffer:] + 880
6  IOGPU                          0x1c5fab930 -[IOGPUMetalCommandEncoder initWithCommandBuffer:] + 52
7  AGXMetalA10                    0x1d4c28a08 (Missing)
8  AGXMetalA10                    0x1d4c871dc (Missing)
9  Flutter                        0x100f97114 (Missing)
10 Flutter                        0x100f9a1c4 (Missing)
11 Flutter                        0x100eabe64 (Missing)
12 Flutter                        0x100eabac4 (Missing)
13 Flutter                        0x100ec9a44 (Missing)
14 Flutter                        0x100eb76d4 (Missing)
15 Flutter                        0x100ed50b4 (Missing)
16 Flutter                        0x100eb72b0 (Missing)
17 Flutter                        0x100e944b4 (Missing)
18 Flutter                        0x100f4caa4 (Missing)
19 Flutter                        0x101093660 (Missing)
20 Flutter                        0x100d2c748 (Missing)
21 Flutter                        0x101092f40 (Missing)
22 Flutter                        0x100d2b42c (Missing)
23 Flutter                        0x100d2d3a0 (Missing)
24 CoreFoundation                 0x18b3ba3e0 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 28
25 CoreFoundation                 0x18b3b9fe4 __CFRunLoopDoTimer + 1004
26 CoreFoundation                 0x18b3b94c4 __CFRunLoopDoTimers + 324
27 CoreFoundation                 0x18b3b3850 __CFRunLoopRun + 1912
28 CoreFoundation                 0x18b3b2ba0 CFRunLoopRunSpecific + 572
29 Flutter                        0x100d2d27c (Missing)
30 Flutter                        0x100d2cc04 (Missing)
31 libsystem_pthread.dylib        0x1d3c5bc9c _pthread_start + 288
32 libsystem_pthread.dylib        0x1d3c60880 thread_start + 8
```

**Comment by dakr0013** [comment]
> I use this method to load images before starting the app, which seems to work as intended:
> 
> `main.dart`
> 
> ```dart
> Future<void> main() async {
>   WidgetsFlutterBinding.ensureInitialized();
>   await loadImage(AssetImage('assets/background-main.jpg'));
>   runApp(MyApp());
> }
> ```
> 
> `preload_image.dart`
> 
> ```dart
> // @dart=2.12
> import 'dart:async';
> 
> import 'package:flutter/foundation.dart';
> import 'package:flutter/painting.dart';
> import 'package:flutter/services.dart';
> 
> Future<void> loadImage(ImageProvider provider) {
>   final config = ImageConfiguration(
>     bundle: rootBundle,
>     devicePixelRatio: 1,
>     platform: defaultTargetPlatform,
>   );
>   final Completer<void> completer = Completer();
>   final ImageStream stream = provider.resolve(config);
> 
>   late final ImageStreamListener listener;
> 
>   listener = ImageStreamListener((ImageInfo image, bool sync) {
>     debugPrint("Image ${image.debugLabel} finished loading");
>     completer.complete();
>     stream.removeListener(listener);
>   }, onError: (dynamic exception, StackTrace? stackTrace) {
>     completer.complete();
>     stream.removeListener(listener);
>     FlutterError.reportError(FlutterErrorDetails(
>       context: ErrorDescription('image failed to load'),
>       library: 'image resource service',
>       exception: exception,
>       stack: stackTrace,
>       silent: true,
>     ));
>   });
> 
>   stream.addListener(listener);
>   return completer.future;
> }
> ```

This didn't work for me because i was using resolution-aware image assets. Because this code always loaded the image for `devicePixelRatio` of 1.0, but my device had a `devicePixelRatio` of 2.0 it still flickered. After changing `devicePixelRatio` to `window.devicePixelRatio` it worked. this is my config: 
```dart
final config = ImageConfiguration(
      bundle: rootBundle,
      devicePixelRatio: window.devicePixelRatio,
      platform: defaultTargetPlatform,
);
```

**Comment by alexbadia1** [comment]
Wow, that's really helpful!

Just to clarify retrieving the devicePixelRatio.... Since you're calling `WidgetsFlutterBinding.ensureInitialized();` you can retrieve the devicePixelRatio from the returned WidgetsBinding instance like so:

```
Future<void> main() async {
    final widgetsBindingInstance = WidgetsFlutterBinding.ensureInitialized();
    await loadImage(
      imageProvider: AssetImage('images/image1.jpg'),
      devicePixelRatio:  widgetsBinding.window.devicePixelRatio,
    );
}// main
```

**Comment by 0x48Maciek** [comment]
I don't know if this would address original issue reporter's problem, but for anyone who stumbles on this issue based on its title, I'd like to share what I learned and yet another solution for preloading images :

Going through methods of WidgetsFlutterBinding I found one called "deferFirstFrame" which is documented the following way:
```dart
/// Tell the framework to not send the first frames to the engine until there
/// is a corresponding call to [allowFirstFrame].
///
/// Call this to perform asynchronous initialization work before the first
/// frame is rendered (which takes down the splash screen). The framework
/// will still do all the work to produce frames, but those frames are never
/// sent to the engine and will not appear on screen.
```

This answered exactly my case of wanting to preload images (and potentially do other things like initialize database or do network requests) before the main app layout is visible, that is, during the splash screen showing.

Here is an example of how I use it:
```dart
Future<void> main() async {
  final binding = WidgetsFlutterBinding.ensureInitialized();

  // Preload all assets to prevent flash when they are loaded.
  binding.deferFirstFrame();
  binding.addPostFrameCallback((_) async {
    BuildContext context = binding.renderViewElement;
    if (context != null) {
      // Run any sync or awaited async function you want to wait for before showing your UI
      await _precacheAssets(context);
    }
    binding.allowFirstFrame();
  });

  runApp(...);
}
```

**Comment by jonbhanson** [comment]
Another solution: Flutter had now implemented [deferFirstFrame](https://api.flutter.dev/flutter/rendering/RendererBinding/deferFirstFrame.html) and (allowFirstFrame)[https://api.flutter.dev/flutter/rendering/RendererBinding/allowFirstFrame.html] which allows the first frame to be hidden until assets are loaded.

My package [flutter_native_splash](https://pub.dev/packages/flutter_native_splash) uses these calls to allow the developer to hide the splash screen only when their app is ready, such as after assets have been loaded.

It seems this question now has sufficient solutions and could be closed.

**Comment by darshankawar** [comment]
Verified this issue using latest stable version by running this code sample and observed that the app launches properly and doesn't show any glitch in first frame:


https://github.com/flutter/flutter/assets/67046386/3fcbdcb8-dba9-4b63-a0c7-aea37ae7efc9


Closing as no longer reproducible. If anybody thinks it is incorrect and still see the originally reported error using latest stable version, please file a new issue with relevant details.
